### PR TITLE
🚧 ✨ Exécuter tâche planifiée

### PR DESCRIPTION
**Pull request by benjlevesque**
A noter:
- le use case "ExécuterTâchePlanifiée" est appelé après l'exécution réelle de la tâche planifiée. Cela permet de s'assurer que l'exécution est un succès, ce qui serait plus complexe si elle était réactive. Cela permet également de découpler les domaines. 
- les specs pour `exécuter` s'affranchissent de GF car le lien tâches<>GF n'est pas testable, puisqu'exécuté dans `scheduled-tasks`
- le scénario de test "Une tâche planifiée est executée même si la date d'execution est dépassée" a été supprimé, ce n'est pas forcément souhaitable. Il existe certains edge cases métier qui pourraient résulter en une erreur d'exécution, mieux vaut monitorer les tâches planifiées et gérer les erreurs, au moins au début

### CODE REVIEW

File: packages/specifications/src/tâche-planifiée/executerTâchePlanifiée.feature [hunk 1]
```diff
@@ -8,22 +8,29 @@ Fonctionnalité: Executer une tâche planifiée
...
         Etant donné des garanties financières validées pour le projet "Du boulodrome de Marseille" avec :
             | type               | avec-date-échéance |
             | date d'échéance    | 2024-12-01         |
             | date de validation | 2024-11-24         |
         Quand on execute les tâches planifiées à la date du "2024-12-02"
-        Alors une tâche "échoir les garanties financières" n'est plus est planifiée pour le projet "Du boulodrome de Marseille"
-        Et les garanties financières du projet "Du boulodrome de Marseilles" sont échues
+        Alors une tâche "échoir les garanties financières" n'est plus planifiée pour le projet "Du boulodrome de Marseille"
 
+    #  Et les garanties financières du projet "Du boulodrome de Marseilles" sont échues
```
Comment (benjlevesque) [inline packages/specifications/src/tâche-planifiée/executerTâchePlanifiée.feature:19]: TODO décommenter

File: packages/specifications/src/tâche-planifiée/stepDefinitions/tâchePlanifiée.when.ts [hunk 2]
```diff
@@ -1,10 +1,58 @@
...
+  async function (
+    this: PotentielWorld,
+    typeTâche: RechercherTypeTâchePlanifiée,
+    nomProjet: string,
+  ) {
+    try {
+      const projet = this.lauréatWorld.rechercherLauréatFixture(nomProjet);
+      const actualTypeTâche = this.tâchePlanifiéeWorld.rechercherTypeTâchePlanifiée(typeTâche);
+
+      // TODO  l'aggregate n'a pas pris en compte les events publiés ?
```
Comment (benjlevesque) [inline packages/specifications/src/tâche-planifiée/stepDefinitions/tâchePlanifiée.when.ts:45]:
```suggestion
```

File: packages/specifications/src/tâche-planifiée/exécuterTâchePlanifiée.feature [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+    # TODO Et les garanties financières du projet "Du boulodrome de Marseilles" sont échues
+    Scénario: Impossible d'executer une tâche planifiée déjà executée
+        Etant donné une tâche executée pour le projet "Du boulodrome de Marseille" avec :
+            | type             | échoir les garanties financières |
+            | date d'exécution | 2024-12-03                       |
+        Quand on execute la tâche planifiée "échoir les garanties financières" pour le projet "Du boulodrome de Marseille"
+        Alors on devrait être informé que "La tâche planifiée est déjà executée"
+
+    # TODO Et les garanties financières du projet "Du boulodrome de Marseilles" sont échues
```
Comment (benjlevesque) [inline packages/specifications/src/tâche-planifiée/exécuterTâchePlanifiée.feature:38]: TODO décommenter

File: packages/specifications/src/tâche-planifiée/exécuterTâchePlanifiée.feature [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    Scénario: Une tâche planifiée est executée
+        # TODO retirer GF ? on teste juste le use case Tacheplanifiée
+        Etant donné des garanties financières validées pour le projet "Du boulodrome de Marseille" avec :
+            | type               | avec-date-échéance |
+            | date d'échéance    | 2024-12-01         |
+            | date de validation | 2024-11-24         |
+        Quand on execute les tâches planifiées à la date du "2024-12-02"
+        Alors une tâche "échoir les garanties financières" n'est plus planifiée pour le projet "Du boulodrome de Marseille"
+
+    # TODO Et les garanties financières du projet "Du boulodrome de Marseilles" sont échues
```
Comment (benjlevesque) [inline packages/specifications/src/tâche-planifiée/exécuterTâchePlanifiée.feature:20]: TODO décommenter

File: packages/specifications/src/tâche-planifiée/exécuterTâchePlanifiée.feature [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+    Contexte:
+        Etant donné le projet lauréat "Du boulodrome de Marseille"
+        Et le porteur pour le projet lauréat "Du boulodrome de Marseille"
+            | email | porteur@test.test   |
+            | nom   | Porteur Projet Test |
+            | role  | porteur-projet      |
+
+    Scénario: Une tâche planifiée est executée
+        # TODO retirer GF ? on teste juste le use case Tacheplanifiée
```
Comment (benjlevesque) [inline packages/specifications/src/tâche-planifiée/exécuterTâchePlanifiée.feature:12]: TODO

File: packages/specifications/src/tâche-planifiée/exécuterTâchePlanifiée.feature [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+    @NotImplemented
+    Scénario: Une tâche planifiée est executée même si la date d'execution est dépassée
+        Etant donné des garanties financières validées pour le projet "Du boulodrome de Marseille" avec :
+            | type               | avec-date-échéance |
+            | date d'échéance    | 2024-12-01         |
+            | date de validation | 2024-11-24         |
+        Quand on execute les tâches planifiées à la date du "2024-12-03"
+        Alors une tâche "échoir les garanties financières" n'est plus planifiée pour le projet "Du boulodrome de Marseille"
+
+    # TODO Et les garanties financières du projet "Du boulodrome de Marseilles" sont échues
```
Comment (benjlevesque) [inline packages/specifications/src/tâche-planifiée/exécuterTâchePlanifiée.feature:30]: TODO

File: packages/domain/tâche-planifiée/src/ajouter/ajouterTâchePlanifiée.behavior.ts [hunk 7]
```diff
@@ -2,22 +2,22 @@ import { IdentifiantProjet, DateTime } from '@potentiel-domain/common';
 import { DomainEvent } from '@potentiel-domain/core';
 
 import * as StatutTâchePlanifiée from '../statutTâchePlanifiée.valueType';
-import * as TypePlanifiéeTâche from '../typeTâchePlanifiée.valueType';
+import * as TâchePlanifiéeType from '../typeTâchePlanifiée.valueType';
```
Comment (HubM) [inline packages/domain/tâche-planifiée/src/ajouter/ajouterTâchePlanifiée.behavior.ts:5]: Dnas les events métiers on utilise le terme `typeTâchePlanifiée`, du coup pourquoi renommer ici?

File: packages/specifications/src/tâche-planifiée/stepDefinitions/tâchePlanifiée.when.ts [hunk 8]
```diff
@@ -1,10 +1,56 @@
...
+      },
+    });
+
+    for (const tâche of tâches.items) {
+      await mediator.send<ExécuterTâchePlanifiéeUseCase>({
+        type: 'System.TâchePlanifiée.UseCase.ExécuterTâchePlanifiée',
+        data: {
+          identifiantProjetValue: tâche.identifiantProjet.formatter(),
+          typeTâchePlanifiéeValue: tâche.typeTâchePlanifiée.type,
+        },
```
Comment (HubM) [inline packages/specifications/src/tâche-planifiée/stepDefinitions/tâchePlanifiée.when.ts:28]: on veut pas trigger l'erreur ici ?

**Review by HubM — CHANGES_REQUESTED** [review]

File: packages/domain/tâche-planifiée/src/ajouter/ajouterTâchePlanifiée.behavior.ts (hunk 7 above)
Comment (benjlevesque) [reply packages/domain/tâche-planifiée/src/ajouter/ajouterTâchePlanifiée.behavior.ts:5]: non c'est une erreur, j'ai voulu corriger la typo (inversion planifiée et tâche) mais tu as raison ca devrait etre `TypeTâchePlanifiée`

File: packages/specifications/src/tâche-planifiée/stepDefinitions/tâchePlanifiée.when.ts (hunk 8 above)
Comment (benjlevesque) [reply packages/specifications/src/tâche-planifiée/stepDefinitions/tâchePlanifiée.when.ts:28]: il faut le mettre à chaque fois le try/catch? même quand ce n'est pas attendu?

Comment (benjlevesque) [reply packages/specifications/src/tâche-planifiée/stepDefinitions/tâchePlanifiée.when.ts]: (je l'ai mis, ca a l'air d'être presque toujours le cas en effet)

**Review by HubM — APPROVED** [review]
